modules/hilary/manifests/init.pp

```diff
@@ -146,17 +146,15 @@
       owner   => $os_user,
       group   => $os_group,
       content => template('hilary/config.js.erb'),
-      require => [ File[$upload_files_dir], File[$config_files_tmp_dir], File[$config_files_tmp_upload_dir] ]
+      require => [ File[$upload_files_dir], File[$config_files_tmp_dir], File[$config_files_tmp_upload_dir] ];
 
     # Env specific config file
     "${app_root_dir}/${environment}.js":
       ensure  => present,
       mode    => "0644",
       owner   => $os_user,
       group   => $os_group,
-      content => template('hilary/${environment}.js'),
-      require => [ File[$upload_files_dir], File[$config_files_tmp_dir], File[$config_files_tmp_upload_dir] ]
-    }
+      source => "puppet:///modules/hilary/${environment}.js";
```

Comment:
Missing closing brace

Reply:
The require line should come back to ensure that the app_root_dir definitely exists;
As app_root_dir is not always under puppet control (eg if hilary is installed via a package) I suggest requiring a file that puppet does control that guarantees the app_root_dir exists;
  require => File[$config_files_tmp_dir]

Reply:
Does the require need to be included in every statement or need to be in the last statement? I left in the original line `require => [ File[$upload_files_dir], File[$config_files_tmp_dir], File[$config_files_tmp_upload_dir] ];` on line 149.